lib/vmdb/settings/validator.rb

```diff
@@ -167,6 +167,42 @@ def smtp(data)
 
         return valid, errors
       end
+
+      def workers(data)
```

Comment:
If I recall, the validator is only presented with the keys that were changed, so we have deal with the data not having both keys.  You might have to fetch the other key from the existing ::Settings section or do some other merge with the default settings before passing that off to `worker_class.worker_settings(data)`.

You should be able to verify that with specs passing in only partial data here.

Reply:
> If I recall, the validator is only presented with the keys that were changed

?  https://github.com/ManageIQ/manageiq/blob/a94268f6aab6d41b691fdb0ea8c07712f357aa61/lib/vmdb/settings/validator.rb#L17-L22

It loops through all keys in the data.

Reply:
Right but that incoming `@config` there might be sparse.

Reply:
Ok, maybe validate was _designed_ to support sparse, but doesn't actually call it that way?  I'll have to dig in with you, because there might be other callers than `.save!`

Reply:
@Fryguy I couldn't find how `@config` could be a partial hash.  Can you double check if you think that's a possibility?  It would be tricky code to deal with only differences.  Plus, the code referenced above calls `validate` before the HashDiffer is used to determine the diffs so I think it's really the complete settings at `validate` time but can't be sure.

Reply:
Yeah, we can review it together.  I'd rather not support something unless it makes sense.